### PR TITLE
adds possibility to upload many observations in one go

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR adds option to upload many observations in one go

How to test resetting a loqusdb link in status db:
1. install on stage (add instructions on how to as needed)
1. find a case that has been uploaded to loqus_db
1. `cg reset observations [case_id]`
1. y

Expected outcome:
The loqusdb-links on the samples for the case have been reset 

How to test resetting all loqusdb links in status db:
1. `cg reset observations`
1. y

Expected outcome:
The loqusdb-links on all samples for all cases that are properly internally linked have been reset 

How to test listing cases that are liable for being uploaded to loqus-db:
1. `cg upload observations -p`

Expected outcome:
All families that has not been uploaded to loqusdb should be listed

How to test uploading all observation in one go:
1. `cg upload observations`

Expected outcome:
All families that has not been uploaded to loqusdb should be uploaded

### CODE REVIEW

File: cg/cli/reset.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import logging
+
+import click
+from cg.store import Store
+LOG = logging.getLogger(__name__)
```
Comment (ingkebil) [inline cg/cli/reset.py]: unused LOG

File: cg/cli/reset.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import logging
```
Comment (ingkebil) [inline cg/cli/reset.py]: unused import

File: cg/cli/reset.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+def observations(context, case_id):
+    """Reset observation links from an analysis to LoqusDB."""
+
+    if case_id:
+        observations_uploaded = [context.obj['status'].family(case_id)]
+    else:
+        observations_uploaded = context.obj['status'].observations_uploaded()
+
+    for family_obj in observations_uploaded:
+        click.echo(click.style(f"This would reset observation links for: {family_obj.internal_id}"))
```
Comment (ingkebil) [inline cg/cli/reset.py]: Maybe we shouldi nvestigate how to use logging for this instead

File: cg/cli/reset.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    click.confirm('Do you want to continue?', abort=True)
+
+    for family_obj in observations_uploaded:
+        context.obj['status'].reset_observations(family_obj.internal_id)
+        click.echo(click.style(f"Reset loqus observations for: {family_obj.internal_id}"))
```
Comment (ingkebil) [inline cg/cli/reset.py]: Maybe move to logging?

File: cg/cli/upload.py [hunk 5]
```diff
@@ -199,42 +199,51 @@ def genotypes(context, re_upload, family_id):
 
 
 @upload.command()
-@click.argument('case_id')
+@click.option('-c', '--case_id', help='internal case id, leave empty to process all')
+@click.option('-p', '--print', 'print_console', is_flag=True, help='only print cases to be processed')
```
Comment (ingkebil) [inline cg/cli/upload.py]: --dry-run

File: cg/cli/reset.py (hunk 1 above)
Comment (patrikgrenfeldt) [reply cg/cli/reset.py]: used :)

File: cg/cli/reset.py (hunk 2 above)
Comment (patrikgrenfeldt) [reply cg/cli/reset.py]: used

File: cg/cli/reset.py (hunk 4 above)
Comment (patrikgrenfeldt) [reply cg/cli/reset.py]: yes

File: cg/cli/upload.py (hunk 5 above)
Comment (patrikgrenfeldt) [reply cg/cli/upload.py]: fixed

File: cg/cli/reset.py (hunk 3 above)
Comment (patrikgrenfeldt) [reply cg/cli/reset.py]: done

File: cg/store/api/status.py [hunk 6]
```diff
@@ -404,6 +404,28 @@ def analyses_to_upload(self):
                                              models.Analysis.uploaded_at == None)
         return records
 
+    def observations_to_upload(self):
+        """Fetch observations that haven't been uploaded."""
+
+        families_q = (
+            self.Family.query
+            .join(models.Analysis, models.Family.links, models.FamilySample.sample)
```
Comment (ingkebil) [inline cg/store/api/status.py]: We might not be using analysis here, right?

**Comment by patrikgrenfeldt** [comment]
@ingkebil we are not using the Analysis object in itself but the existence
indicates that an Analysis has been performed and I suppose also that
observations are ready to be uploaded. Do you have any other suggestion on
how we can know that we are ready to upload?

Den tors 11 apr. 2019 14:53Kenny Billiau <notifications@github.com> skrev:

**Comment by ingkebil** [comment]
Alright! Looking good then.

**Pull request merged**